test/e2e/upgrade_management_cluster_test.go

```diff
@@ -35,307 +24,128 @@ func upgradeManagementCluster() {
-				client.InNamespace(namespace),
-				client.MatchingLabels{clusterv1.ClusterLabelName: clusterName},
-			}
-			g.Expect(bootstrapClusterProxy.GetClient().List(ctx, machineList, opts...)).To(Succeed())
-			for _, machine := range machineList.Items {
-				if machine.Status.GetTypedPhase() == clusterv1alpha4.MachinePhaseRunning {
-					n++
-				}
-			}
-			g.Expect(n).To(Equal(2))
-		}, e2eConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Succeed())
-
-		upgradeClusterProxy := bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace, clusterName)
-		upgradeClusterClient := upgradeClusterProxy.GetClient()
-
-		By("Initializing the workload cluster with older versions of providers")
-
-		contract := e2eConfig.GetVariable("CAPI_VERSION")
-
-		clusterctl.InitManagementClusterAndWatchControllerLogs(ctx, clusterctl.InitManagementClusterAndWatchControllerLogsInput{
-			ClusterctlBinaryPath:    clusterctlBinaryPath, // use older version of clusterctl to init the management cluster
-			ClusterProxy:            upgradeClusterProxy,
-			ClusterctlConfigPath:    clusterctlConfigPath,
-			CoreProvider:            e2eConfig.GetProviderLatestVersionsByContract(contract, config.ClusterAPIProviderName)[0],
-			BootstrapProviders:      e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmBootstrapProviderName),
-			ControlPlaneProviders:   e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmControlPlaneProviderName),
-			InfrastructureProviders: e2eConfig.GetProviderLatestVersionsByContract(contract, e2eConfig.InfrastructureProviders()...),
-			LogFolder:               filepath.Join(artifactFolder, "clusters", clusterName),
-		}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
-
-		/*-------------------------------------------------*
-		| Pivot to run ironic/BMO and resources on target |
-		*--------------------------------------------------*/
-		By("Start pivoting")
-		listBareMetalHosts(ctx, bootstrapClusterProxy.GetClient(), client.InNamespace(namespace))
-		listNodes(ctx, upgradeClusterClient)
-
-		By("Remove Ironic containers from the source cluster")
-		ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
-		if ephemeralCluster == KIND {
-			removeIronicContainers()
-		} else {
-			removeIronicDeployment()
-		}
-
-		By("Create Ironic namespace")
-		upgradeClusterClientSet := upgradeClusterProxy.GetClientSet()
-		ironicNamespace := &corev1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: e2eConfig.GetVariable("IRONIC_NAMESPACE"),
-			},
+	capi_e2e.ClusterctlUpgradeSpec(ctx, func() capi_e2e.ClusterctlUpgradeSpecInput {
+		return capi_e2e.ClusterctlUpgradeSpecInput{
+			E2EConfig:                 e2eConfig,
+			ClusterctlConfigPath:      clusterctlConfigPath,
+			BootstrapClusterProxy:     bootstrapClusterProxy,
+			ArtifactFolder:            artifactFolder,
+			SkipCleanup:               true, // TODO(mboukhalfa): Parameterize after merging https://github.com/kubernetes-sigs/cluster-api/pull/7373
```

Comment:
linked PR is merged, can you do the parameterization now?

Reply:
We should wait for the new patch release